src/connectors/geo-search/connectGeoSearch.js

```diff
@@ -192,9 +193,32 @@ const connectGeoSearch = (renderFn, unmountFn) => {
     const hasMapMoveSinceLastRefine = () =>
       widgetState.hasMapMoveSinceLastRefine;
 
+    const getSearchBounds = helper => () => {
+      const boundingBox =
+        helper.state.insideBoundingBox &&
+        helper.state.insideBoundingBox.split(',');
+
+      if (!boundingBox) {
+        return undefined;
+      }
+
+      return {
+        northEast: {
+          lat: parseFloat(boundingBox[0]),
+          lng: parseFloat(boundingBox[1]),
+        },
+        southWest: {
+          lat: parseFloat(boundingBox[2]),
+          lng: parseFloat(boundingBox[3]),
+        },
+      };
+    };
+    let getSearchBoundsInstance;
+
     const init = initArgs => {
       const { state, helper, instantSearchInstance } = initArgs;
       const isFirstRendering = true;
+      getSearchBoundsInstance = getSearchBounds(helper);
```

Comment:
I don't think we need to save the instance. We keep the exact same instance only for the function that will trigger a new render (because otherwise we have stalled values in the next render). This function works like `refine` or `clearMapRefinement`.

Reply:
The Helper is mutable and will always be up to date. It's not great to generate at each render the exact same function. In other widgets we try to be careful about those, and we cache the refine functions.

Reply:
In which context it's not great? For performance?

Reply:
Since we don't have performance issue I prefer to go for the more simple way, that is (IMO) to create the function every time. Indeed we pay some performance costs but we don't have any issue regarding performance for now. At the end both solutions work, I'm fine with it. Just let's be consistent across the widget.